07week/towersofhanoi-app/src/App.js

```diff
@@ -0,0 +1,71 @@
+// https://github.com/deytonk/javascript-workbook/pull/17?
+
+import React, { Component } from 'react';
+import logo from './logo.svg';
+import './App.css';
+import TowersButton from './TowersButton';
+
+class App extends Component {
+  state = {
+    stacks: {
+      a: [4, 3, 2, 1],
+      b: [],
+      c: []
+    },
+    stackColorObj: {
+      4: 'red',
+      3: 'orange',
+      2: 'yellow',
+      1: 'green'
+    },
+    startStack: null,
+    endStack: null
+  }
+
+  isLegalMove(startStack, endStack) {
+    const endValue = this.state.stacks[endStack][this.state.stacks[endStack].length];
+    const startValue = this.state.stacks[this.state.startStack][this.state.stacks[startStack].length];
+    if(!startValue) {
+      return "This stack is empty! Try again.";
+    } else if(!endValue) {
+      return true;
+    } else if (endValue > startValue) {
+      return "Illegal move! Try again.";
+    }
+  }
+
+  handleStackClick(startStack, endStack) {
```

Comment:
Here, you use methods assuming an endStack, but you never pass one in. Which doesn't really make sense, since you only click one stack at a time.